src/Core/Variables/StandardVariableProvider.php

```diff
@@ -356,20 +356,27 @@ protected function detectAccessor($subject, $propertyName)
         }
         if (is_object($subject)) {
             $className = $this->populateGettersByClassName($subject);
-            $upperCasePropertyName = ucfirst($propertyName);
-            $getter = 'get' . $upperCasePropertyName;
+            $lowerCasePropertyName = strtolower($propertyName);
+            $getter = 'get' . $lowerCasePropertyName;
             if (static::$gettersByClassName[$className][$getter] ?? false) {
                 return self::ACCESSOR_GETTER;
             }
             if ($this->isExtractableThroughAsserter($subject, $propertyName)) {
                 return self::ACCESSOR_ASSERTER;
             }
-            if (property_exists($subject, $propertyName)) {
+            $propertyNames = array_keys(get_object_vars($subject));
```

Comment:
I would really like to avoid this one - the code here executes thousands and thousands of times in a normal set of Fluid templates, so if we every time extract array keys and get object vars for the same object, that will add a lot of pressure. Perhaps it can be done by the method that populates the array of getters per class name.